unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NotificationMessagesHUD/MainChatNotificationsComponentView.cs

```diff
@@ -0,0 +1,273 @@
+    }
+
+    private void SetScrollToEnd()
+    {
+        scrollRectangle.normalizedPosition = new Vector2(0, 0);
+        ResetNotificationButton();
+    }
+
+    private void PopulatePrivateNotification(ChatNotificationMessageComponentView chatNotificationComponentView, ChatMessage message, string username = null, string profilePicture = null)
+    {
+        chatNotificationComponentView.SetIsPrivate(true);
+        chatNotificationComponentView.SetMessage(message.body);
+        chatNotificationComponentView.SetNotificationHeader("Private message");
+        chatNotificationComponentView.SetNotificationSender(username);
+        chatNotificationComponentView.SetNotificationTargetId(message.sender);
+        if (profilePicture != null)
+            chatNotificationComponentView.SetImage(profilePicture);
+    }
+
+    private void PopulatePublicNotification(ChatNotificationMessageComponentView chatNotificationComponentView, ChatMessage message, string username = null)
+    {
+        chatNotificationComponentView.SetIsPrivate(false);
+        chatNotificationComponentView.SetMessage(message.body);
+        chatNotificationComponentView.SetNotificationTargetId("#nearby");
+        chatNotificationComponentView.SetNotificationHeader("#nearby");
+        chatNotificationComponentView.SetNotificationSender(username);
+    }
+
+    private void ClickedOnNotification(string targetId)
+    {
+        OnClickedNotification?.Invoke(targetId);
+    }
+
+    private void FocusedOnNotification(bool isInFocus)
+    {
+        isOverMessage = isInFocus;
+        OnResetFade?.Invoke(!isOverMessage && !isOverPanel);
+    }
+
+    private void FocusedOnPanel(bool isInFocus)
+    {
+        isOverPanel = isInFocus;
+        OnPanelFocus?.Invoke(isOverPanel);
+        OnResetFade?.Invoke(!isOverMessage && !isOverPanel);
+    }
+
+    private void CheckNotificationCountAndRelease()
+    {
+        if (poolableQueue.Count >= MAX_NOTIFICATION_ENTRIES)
+        {
+            ChatNotificationMessageComponentView notificationToDequeue = notificationQueue.Dequeue(); 
+            notificationToDequeue.onFocused -= FocusedOnNotification;
+            entryPool.Release(poolableQueue.Dequeue());
+        }
+    }
+
+    private Pool GetNotificationEntryPool()
+    {
+        var entryPool = PoolManager.i.GetPool(NOTIFICATION_POOL_NAME_PREFIX + name + GetInstanceID());
+        if (entryPool != null) return entryPool;
+
+        entryPool = PoolManager.i.AddPool(
+            NOTIFICATION_POOL_NAME_PREFIX + name + GetInstanceID(),
+            Instantiate(chatNotification).gameObject,
+            maxPrewarmCount: MAX_NOTIFICATION_ENTRIES,
+            isPersistent: true);
+        entryPool.ForcePrewarm();
+
+        return entryPool;
+    }
+
+    private static string UnixTimeStampToLocalTime(ulong unixTimeStampMilliseconds)
+    {
+        DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
```

Comment:
you can replace with: `DateTimeOffset.FromUnixTimeMilliseconds(unixTimeStampMilliseconds)`

Reply:
I'm not managing to